Mathlib/Analysis/SpecialFunctions/Pow/Complex.lean

```diff
@@ -112,55 +112,95 @@ theorem cpow_sub {x : ℂ} (y z : ℂ) (hx : x ≠ 0) : x ^ (y - z) = x ^ y / x
 theorem cpow_neg_one (x : ℂ) : x ^ (-1 : ℂ) = x⁻¹ := by simpa using cpow_neg x 1
 #align complex.cpow_neg_one Complex.cpow_neg_one
 
+/-- See also `Complex.cpow_int_mul'`. -/
+lemma cpow_int_mul (x : ℂ) (n : ℤ) (y : ℂ) : x ^ (n * y) = (x ^ y) ^ n := by
+  rcases eq_or_ne x 0 with rfl | hx
+  · rcases eq_or_ne n 0 with rfl | hn
+    · simp
+    · rcases eq_or_ne y 0 with rfl | hy <;> simp [*, zero_zpow]
+  · rw [cpow_def_of_ne_zero hx, cpow_def_of_ne_zero hx, mul_left_comm, exp_int_mul]
+
+lemma cpow_mul_int (x y : ℂ) (n : ℤ) : x ^ (y * n) = (x ^ y) ^ n := by rw [mul_comm, cpow_int_mul]
+
+lemma cpow_nat_mul (x : ℂ) (n : ℕ) (y : ℂ) : x ^ (n * y) = (x ^ y) ^ n := by
+  exact_mod_cast cpow_int_mul x n y
+
+lemma cpow_ofNat_mul (x : ℂ) (n : ℕ) [n.AtLeastTwo] (y : ℂ) :
+    x ^ (OfNat.ofNat n * y) = (x ^ y) ^ (OfNat.ofNat n : ℕ) :=
+  cpow_nat_mul x n y
```

Comment:
These probably need `no_index` too; can you reference the library note at each `ofNat` lemma? (it's right next to where `AtLeastTwo` is defined)